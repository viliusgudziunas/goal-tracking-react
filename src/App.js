import React, { useState, useEffect } from 'react';
import Goals from './components/Goals';
import GoalForm from './components/GoalForm';
import GoalsHeader from './components/GoalsHeader';
import validationService from './components/services/validationService';

function App() {
  const [goals, setGoals] = useState([]);

  useEffect(() => {
    fetch('/users/1').then(res =>
      res.json().then(data => {
        setGoals(data);
      })
    );
  }, []);

  const onCompleteGoal = updatedGoal => {
    goals.splice(goals.indexOf(updatedGoal), 1, updatedGoal);
    setGoals([...goals]);
  };

  const onChangeTarget = updatedGoal => {
    goals.splice(goals.indexOf(updatedGoal), 1, updatedGoal);
    setGoals([...goals]);
  };

  const validateNewGoalName = newGoalName => {
    return validationService.goalNameValidation(goals, newGoalName);
  };

  return (
    <div>
      <GoalsHeader />
      <Goals
        goals={goals}
        onDeleteGoal={updatedGoals => setGoals([...updatedGoals])}
        onCompleteGoal={onCompleteGoal}
        onChangeTarget={onChangeTarget}
      />
      <GoalForm
        onNewGoal={newGoal => setGoals([...goals, newGoal])}
        validateNewGoalName={validateNewGoalName}
      />
    </div>
  );
}

export default App;
