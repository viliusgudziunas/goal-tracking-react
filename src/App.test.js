import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashed', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
