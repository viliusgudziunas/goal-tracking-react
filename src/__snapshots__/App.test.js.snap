// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashed 1`] = `
<div>
  <GoalsHeader />
  <Goals
    goals={Array []}
    onChangeTarget={[Function]}
    onCompleteGoal={[Function]}
    onDeleteGoal={[Function]}
  />
  <GoalForm
    onNewGoal={[Function]}
    validateNewGoalName={[Function]}
  />
</div>
`;
